### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'buffer_id'

### ISSUE THREAD

**Issue by gerardroche** [SublimeLinter/SublimeLinter#1393]
Sometimes during startup I get this exception:

```
Traceback (most recent call last):
  File "/home/code/sublime_text_3/sublime_plugin.py", line 594, in on_selection_modified_async
    callback.on_selection_modified_async(v)
  File "/home/code/.config/sublime-text-3/Packages/SublimeLinter/highlight_view.py", line 247, in on_selection_modified_async
    if view.buffer_id() == active_view.buffer_id():
AttributeError: 'NoneType' object has no attribute 'buffer_id'
```

I looked into it a little and found that the `State['active_view']` doesn't have the active view in `"/home/code/.config/sublime-text-3/Packages/SublimeLinter/highlight_view.py", line 247`. I couldn't figure out why. It happens ~35% of the time.

**Comment by kaste** [comment]
Hi! I hoped these startup/init errors would go away with 3.1. Are you on a new ST version?

**Comment by gerardroche** [comment]
Yep.

- SL 4.5.0
- Sublime Text 3170
- Platform: linux
- Arch: x64
- Install via PC: False

**Comment by kaste** [comment]
Hooray!

**Comment by gerardroche** [comment]
Here a few things I've noticed when trying to debug it:

There is another similar excaption:

```
Traceback (most recent call last):
  File "/home/code/sublime_text_3/sublime_plugin.py", line 283, in on_api_ready
    m.plugin_loaded()
  File "/home/code/.config/sublime-text-3/Packages/SublimeLinter/sublime_linter.py", line 55, in plugin_loaded
    hit(view)
  File "/home/code/.config/sublime-text-3/Packages/SublimeLinter/sublime_linter.py", line 245, in hit
    bid = view.buffer_id()
AttributeError: 'NoneType' object has no attribute 'buffer_id'
```

I think that is caused when one of the active views is an image. `active_view()` will return `None` for things like images. I think `active_sheet()` is used for things like images. So the `hit()` here https://github.com/SublimeLinter/SublimeLinter/blob/master/sublime_linter.py#L49 is receiving None.

That is not causing the other issue, but may be compounding it.

There seems to be a race-condition for the `plugin_loaded()` functions. Sometimes `highlight_view.plugin_loaded()` is loaded first and sometimes `sublime_linter.plugin_loaded()` is loaded first. I think when `highlight_view.plugin_loaded()` loads first the exception is raised. This may be why the issue only occurs some of the time.

**Comment by kaste** [comment]
You have a patched `highlight_view.py`. Line 247 does not contain that code.

Thanks for the second traceback. Fixed in #1395
